src/rgw/driver/rados/rgw_rest_user.cc

```diff
@@ -308,6 +312,16 @@ void RGWOp_User_Modify::execute(optional_yield y)
   RESTArgs::get_string(s, "default-placement", default_placement_str, &default_placement_str);
   RESTArgs::get_string(s, "placement-tags", placement_tags_str, &placement_tags_str);
 
+  // if keys cap is not set to write a keys cannot be generated
+  if(gen_key && (s->user->get_info().caps.check_cap("keys", RGW_CAP_WRITE) == -EPERM)) {
```

Comment:
@cbodley What would you think about removing these parameter specific checks and just having the op (in this instance modify user) return -EPERM if `keys` are not set to `write` in `check_caps` just like I'm doing in `RGWOp_User_Create`. `RGWOp_Subuser_Modify` is another example of this.

Reply:
we shouldn't prevent someone with `user=write` caps from modifying other fields of the user info. you're generally doing the right thing with these parameter-specific checks

i wish we could do that in the `check_caps()` function, but it's not really worth restructuring all of these ops in order to parse their params before `verify_permissions()`/`check_caps()`